lib/Ast.ml

```diff
@@ -313,6 +313,8 @@ module Exp = struct
     | Pexp_apply
         ({pexp_desc= Pexp_ident {txt= Lident "not"; _}; _}, [(_, e1)]) ->
         is_trivial e1
+    | Pexp_variant (_, None) -> true
+    | Pexp_list x | Pexp_array x -> List.for_all x ~f:is_trivial
```

Comment:
"trivial" is stronger than "simple". I think we shouldn't consider lists and arrays to be "trivial".

What do you think of these rules:
- list/array is not trivial
- list/array containing only trivial elements is simple (instead of trivial here)

Reply:
This distinction makes sense. It was already done in `Ast.is_simple` to have "a list/array containing only trivial elements is simple". I updated the PR for now.
 @ceastlund @cwong-ocaml what do you think about this distinction?

Reply:
I know nothing about what this distinction means. But if the "~f:(fun ...)" can still be on the same line with it, I think it's fine.

Reply:
With this distinction,
this list is simple because it's only compose of trivial elements so we do not break between the list and the following argument:
```
  foo [ A; B ] ~f:(fun thing ->
      something that reaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaally needs wrapping)
```
this list is composed of non trivial elements (the list), so there is a break between the list and the following argument:
```
  foo
    [ [ A ]; B ]
    ~f:(fun thing ->
      something that reaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaally needs wrapping)
```

This distinction is purely arbitrary but we have to draw the line somewhere to decide what to wrap/what to break. @ceastlund I guess on realistic code most non-simple lists should not fit on a single line anyway, do you really have a big % of non-simple ones that fit like the one of the example?

Reply:
In my head, I would not normally make such a distinction. If something fits on one line, I'd probably be happy to squeeze it between "foo" and "~f". And if I had to think of exceptions to that, my highest candidates would probably be control operators like if/then/else. I wouldn't worry about data constructors, even nested ones, whether lists or tuples or variants or records. There's no purely data expression I'd object to fitting on the same line.

Reply:
Here is the diff if we just ignore the contents of the data:

conventional profile:
<details>

```diff
   assert_has_nullability_info storage
     { class_name = "a.A"; method_name = Method "foo"; param_types = [ "b.B" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"test.sig" ~expected_line:1;
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"test.sig"
+    ~expected_line:1;
   assert_has_nullability_info storage
     {
       class_name = "b.B";
@@ -133,8 +133,8 @@ let disregards_whitespace_lines_and_comments =
   in
   assert_has_nullability_info storage
     { class_name = "a.A"; method_name = Method "foo"; param_types = [ "b.B" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"test.sig" ~expected_line:2;
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"test.sig"
+    ~expected_line:2;
   (* Commented out signatures should be ignored *)
   assert_no_info storage
     { class_name = "a.A"; method_name = Method "bar"; param_types = [ "b.B" ] }
@@ -260,8 +260,8 @@ let can_add_several_files =
   in
   assert_has_nullability_info storage
     { class_name = "a.A"; method_name = Method "foo"; param_types = [ "b.B" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"file1.sig" ~expected_line:1;
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"file1.sig"
+    ~expected_line:1;
   (* 2. Add another file and check if we added info *)
   let file2 = [ "e.E#baz(f.F)"; "g.G#<init>(h.H, @Nullable i.I) @Nullable" ] in
   let storage =
@@ -270,13 +270,13 @@ let can_add_several_files =
   in
   assert_has_nullability_info storage
     { class_name = "e.E"; method_name = Method "baz"; param_types = [ "f.F" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"file2.sig" ~expected_line:1;
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"file2.sig"
+    ~expected_line:1;
   (* 3. Ensure we did not forget the content from the first file *)
   assert_has_nullability_info storage
     { class_name = "a.A"; method_name = Method "foo"; param_types = [ "b.B" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"file1.sig" ~expected_line:1
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"file1.sig"
+    ~expected_line:1
 
 let should_not_forgive_unparsable_strings =
   "should_not_forgive_unparsable_strings" >:: fun _ ->
diff --git a/infer/src/pulse/PulseCallOperations.ml b/infer/src/pulse/PulseCallOperations.ml
index ff767ed62..52901bca9 100644
--- a/infer/src/pulse/PulseCallOperations.ml
+++ b/infer/src/pulse/PulseCallOperations.ml
@@ -37,8 +37,7 @@ let unknown_call tenv path call_loc (reason : CallEvent.t) ~ret ~actuals
             List.fold fields ~init:astate ~f:(fun acc (field, field_typ, _) ->
                 let fresh_value = AbstractValue.mk_fresh () in
                 Memory.add_edge addr (FieldAccess field)
-                  (fresh_value, [ event ])
-                  call_loc acc
+                  (fresh_value, [ event ]) call_loc acc
                 |> havoc_fields (fresh_value, history) field_typ)
         | None -> astate)
     | _ -> astate
@@ -57,9 +56,8 @@ let unknown_call tenv path call_loc (reason : CallEvent.t) ~ret ~actuals
            raising issues when havoc'ing pointer parameters (which normally causes a [check_valid]
            call). *)
         let fresh_value = AbstractValue.mk_fresh () in
-        Memory.add_edge actual Dereference
-          (fresh_value, [ event ])
-          call_loc astate
+        Memory.add_edge actual Dereference (fresh_value, [ event ]) call_loc
+          astate
         |> havoc_fields actual typ
     | _ -> astate
   in
diff --git a/infer/src/pulse/PulseFormula.ml b/infer/src/pulse/PulseFormula.ml
index 10ec58783..5a6256100 100644
--- a/infer/src/pulse/PulseFormula.ml
+++ b/infer/src/pulse/PulseFormula.ml
@@ -170,8 +170,7 @@ end = struct
           | VarSubst v when Var.equal v v_foreign -> ()
           | _ -> changed := true);
           (acc_f, add (mult q0 (of_subst_target op)) l))
-        vs_foreign
-        (init, (c, Var.Map.empty))
+        vs_foreign (init, (c, Var.Map.empty))
     in
     let l' = if !changed then l' else l0 in
     (acc_f, l')
diff --git a/infer/src/pulse/PulseModels.ml b/infer/src/pulse/PulseModels.ml
index 2a702aaf4..fb60e3e9b 100644
--- a/infer/src/pulse/PulseModels.ml
+++ b/infer/src/pulse/PulseModels.ml
@@ -157,8 +157,7 @@ module Misc = struct
               class_name;
             PulseCallOperations.call tenv path ~caller_proc_desc:proc_desc
               ~callee_data:(analyze_dependency destructor)
-              location destructor ~ret
-              ~actuals:[ (deleted_access, typ) ]
+              location destructor ~ret ~actuals:[ (deleted_access, typ) ]
               ~formals_opt:None astate)
     | _ ->
         L.d_printfln
@@ -380,8 +379,7 @@ module ObjC = struct
     let<+> astate, _ =
       PulseOperations.eval_access path
         ~must_be_valid_reason:InsertionIntoCollectionKey Read location
-        (key, event :: key_hist)
-        Dereference astate
+        (key, event :: key_hist) Dereference astate
     in
     astate
 
@@ -712,8 +710,7 @@ module StdAtomicInteger = struct
     in
     let+ astate =
       PulseOperations.write_deref path location ~ref:int_addr
-        ~obj:(bop_addr, event :: hist)
-        astate
+        ~obj:(bop_addr, event :: hist) astate
     in
     let ret_int = match prepost with `Pre -> bop_addr | `Post -> old_int in
     PulseOperations.write_id ret_id (ret_int, event :: hist) astate
@@ -898,8 +895,7 @@ module StdBasicString = struct
         astate
     in
     let<+> astate =
-      PulseOperations.write_field path location
-        ~ref:(addr, event :: hist)
+      PulseOperations.write_field path location ~ref:(addr, event :: hist)
         internal_string ~obj:init_hist astate
     in
     astate
@@ -1037,8 +1033,7 @@ module StdFunction = struct
       let empty_target = AbstractValue.mk_fresh () in
       let<+> astate =
         PulseOperations.write_deref path location ~ref:dest
-          ~obj:(empty_target, [ event ])
-          astate
+          ~obj:(empty_target, [ event ]) astate
       in
       PulseOperations.havoc_id ret_id [ event ] astate
     else
@@ -1165,8 +1160,7 @@ module GenericArrayBackedCollectionIterator = struct
       to_internal_pointer path Read location init astate
     in
     PulseOperations.write_field path location ~ref internal_pointer
-      ~obj:(p_addr, event :: p_hist)
-      astate
+      ~obj:(p_addr, event :: p_hist) astate
 
   let constructor ~desc this init : model =
    fun { path; location } astate ->
@@ -1505,14 +1499,12 @@ module JavaCollection = struct
     let init_value = AbstractValue.mk_fresh () in
     (* The two internal fields are initially set to null *)
     let<*> astate =
-      Java.write_field path fst_field
-        (init_value, [ event ])
-        location fresh_val astate
+      Java.write_field path fst_field (init_value, [ event ]) location fresh_val
+        astate
     in
     let<*> astate =
-      Java.write_field path snd_field
-        (init_value, [ event ])
-        location fresh_val astate
+      Java.write_field path snd_field (init_value, [ event ]) location fresh_val
+        astate
     in
     (* The empty field is initially set to true *)
     let<*> astate =
@@ -1581,9 +1573,8 @@ module JavaCollection = struct
     in
     let is_empty_val = AbstractValue.mk_fresh () in
     let<*> astate' =
-      Java.write_field path is_empty_field
-        (is_empty_val, [ event ])
-        location coll_val astate
+      Java.write_field path is_empty_field (is_empty_val, [ event ]) location
+        coll_val astate
     in
     (* case1: fst_field is updated *)
     let astate1 =
@@ -1626,9 +1617,8 @@ module JavaCollection = struct
     let ret_val = AbstractValue.mk_fresh () in
     let is_empty_val = AbstractValue.mk_fresh () in
     let* astate =
-      Java.write_field path is_empty_field
-        (is_empty_val, [ event ])
-        location coll_val astate
+      Java.write_field path is_empty_field (is_empty_val, [ event ]) location
+        coll_val astate
     in
     let* astate =
       PulseArithmetic.and_eq_int null_val IntLit.zero astate
@@ -1640,8 +1630,7 @@ module JavaCollection = struct
     in
     let+ astate =
       PulseOperations.write_deref path location ~ref:field_addr
-        ~obj:(null_val, [ event ])
-        astate
+        ~obj:(null_val, [ event ]) astate
     in
     PulseOperations.write_id ret_id (ret_val, [ event ]) astate
 
@@ -1715,19 +1704,16 @@ module JavaCollection = struct
       PulseOperations.eval_access path Read location coll Dereference astate
     in
     let<*> astate =
-      Java.write_field path fst_field
-        (null_val, [ event ])
-        location coll_val astate
+      Java.write_field path fst_field (null_val, [ event ]) location coll_val
+        astate
     in
     let<*> astate =
-      Java.write_field path snd_field
-        (null_val, [ event ])
-        location coll_val astate
+      Java.write_field path snd_field (null_val, [ event ]) location coll_val
+        astate
     in
     let<*> astate =
-      Java.write_field path is_empty_field
-        (is_empty_val, [ event ])
-        location coll_val astate
+      Java.write_field path is_empty_field (is_empty_val, [ event ]) location
+        coll_val astate
     in
     let<+> astate =
       PulseArithmetic.and_eq_int null_val IntLit.zero astate
@@ -1752,8 +1738,7 @@ module JavaCollection = struct
     let astate = PulseOperations.write_id ret_id (not_found_val, hist) astate in
     PulseOperations.invalidate path
       (StackAddress (Var.of_id ret_id, hist))
-      location (ConstantDereference IntLit.zero)
-      (not_found_val, [ event ])
+      location (ConstantDereference IntLit.zero) (not_found_val, [ event ])
       astate
 
   (* Auxiliary function that splits the state into three, considering the case that
diff --git a/sledge/src/fol/arithmetic.ml b/sledge/src/fol/arithmetic.ml
index dedc91088..c2954c27c 100644
--- a/sledge/src/fol/arithmetic.ml
+++ b/sledge/src/fol/arithmetic.ml
@@ -342,8 +342,7 @@ struct
       (match get_mono poly with
       | Some mono ->
           let mono', (coeff, mono_delta) =
-            Prod.fold mono
-              (mono, (Q.one, Mono.one))
+            Prod.fold mono (mono, (Q.one, Mono.one))
               ~f:(fun base power (mono', delta) ->
                 let base' = f base in
                 if base' == base then (mono', delta)
diff --git a/sledge/src/fol/context.ml b/sledge/src/fol/context.ml
index d10772687..1ef84e12f 100644
--- a/sledge/src/fol/context.ml
+++ b/sledge/src/fol/context.ml
@@ -758,9 +758,7 @@ let inter wrt r s =
   else
     let merge_mems rs r s =
       Trm.Map.fold s.cls rs ~f:(fun ~key:rep ~data:cls rs ->
-          Cls.fold cls
-            ([ rep ], rs)
-            ~f:(fun exp (reps, rs) ->
+          Cls.fold cls ([ rep ], rs) ~f:(fun exp (reps, rs) ->
               match
                 List.find ~f:(fun rep -> implies r (Fml.eq exp rep)) reps
               with
diff --git a/sledge/src/sh.ml b/sledge/src/sh.ml
index 9b2b4a397..982960ba8 100644
--- a/sledge/src/sh.ml
+++ b/sledge/src/sh.ml
@@ -611,8 +611,7 @@ let fold_dnf ~conj ~disj sjn (xs, conjuncts) disjuncts =
     let sjn = { sjn with djns = [] } in
     split_case
       (Iter.append (Iter.of_list djns) pending_splits)
-      (xs, conj sjn conjuncts)
-      disjuncts
+      (xs, conj sjn conjuncts) disjuncts
   and split_case pending_splits (xs, conjuncts) disjuncts =
     match Iter.pop pending_splits with
     | Some (split, pending_splits) ->
diff --git a/compiler/lib/generate.ml b/compiler/lib/generate.ml
index b3a8e9b54e..0d0f3e1c01 100644
--- a/compiler/lib/generate.ml
+++ b/compiler/lib/generate.ml
@@ -414,8 +414,7 @@ let access_queue' ~ctx queue x =
 let access_queue_may_flush queue v x =
   let tx, queue = access_queue queue x in
   let _, instrs, queue =
-    List.fold_left queue
-      ~init:(Code.Var.Set.singleton v, [], [])
+    List.fold_left queue ~init:(Code.Var.Set.singleton v, [], [])
       ~f:(fun (deps, instrs, queue) ((y, elt) as eq) ->
         if Code.Var.Set.exists (fun p -> Code.Var.Set.mem p deps) elt.deps then
           ( Code.Var.Set.add y deps,
@@ -962,8 +961,7 @@ let throw_statement ctx cx k loc =
         ( J.Throw_statement
             (ecall
                (runtime_fun ctx "caml_exn_with_js_backtrace")
-               [ cx; bool (int 1) ]
-               loc),
+               [ cx; bool (int 1) ] loc),
           loc );
       ]
   | `Reraise ->
@@ -971,8 +969,7 @@ let throw_statement ctx cx k loc =
         ( J.Throw_statement
             (ecall
                (runtime_fun ctx "caml_exn_with_js_backtrace")
-               [ cx; bool (int 0) ]
-               loc),
+               [ cx; bool (int 0) ] loc),
           loc );
       ]
 
diff --git a/compiler/lib/generate_closure.ml b/compiler/lib/generate_closure.ml
index 25ea3e6666..ef59bd19d3 100644
--- a/compiler/lib/generate_closure.ml
+++ b/compiler/lib/generate_closure.ml
@@ -446,10 +446,8 @@ let rec rewrite_closures mutated_vars rewrite_list free_pc blocks body :
                          closures_map)
                     |> Array.to_list
               in
-              List.fold_left
-                ~init:(free_pc, blocks, { int = []; ext = [] })
-                components
-                ~f:(fun (free_pc, blocks, acc) component ->
+              List.fold_left ~init:(free_pc, blocks, { int = []; ext = [] })
+                components ~f:(fun (free_pc, blocks, acc) component ->
                   let free_pc, blocks, ie =
                     rewrite_tc free_pc blocks closures_map component
                   in
diff --git a/compiler/lib/inline.ml b/compiler/lib/inline.ml
index 0ac0f5aebe..b8562c3dda 100644
--- a/compiler/lib/inline.ml
+++ b/compiler/lib/inline.ml
@@ -180,8 +180,7 @@ let rec args_equal xs ys =
 let inline closures live_vars outer_optimizable pc (blocks, free_pc) =
   let block = Addr.Map.find pc blocks in
   let body, (branch, blocks, free_pc) =
-    List.fold_right block.body
-      ~init:([], (block.branch, blocks, free_pc))
+    List.fold_right block.body ~init:([], (block.branch, blocks, free_pc))
       ~f:(fun i (rem, state) ->
         match i with
         | Let (x, Apply (f, args, true)) when Var.Map.mem f closures -> (
diff --git a/compiler/lib/linker.ml b/compiler/lib/linker.ml
index 31a15239bb..9dfc700c7e 100644
--- a/compiler/lib/linker.ml
+++ b/compiler/lib/linker.ml
@@ -101,8 +101,7 @@ let parse_from_lex ~filename lex =
         else (acc, x :: l)
   in
   let status, blocks, _comments =
-    List.fold_left program
-      ~init:(`Annot [], [], comments)
+    List.fold_left program ~init:(`Annot [], [], comments)
       ~f:(fun (status, blocks, comments) t ->
         match t with
         | _, Javascript.Pi loc ->
diff --git a/compiler/tests-ocaml/basic-modules/recursive_module_init.ml b/compiler/tests-ocaml/basic-modules/recursive_module_init.ml
index f3e08baef7..00e8f9c131 100644
--- a/compiler/tests-ocaml/basic-modules/recursive_module_init.ml
+++ b/compiler/tests-ocaml/basic-modules/recursive_module_init.ml
@@ -8,9 +8,7 @@ let check ~stub txt f =
   in
   Printf.printf "%5s[%s]: nonrec => %s, self => %s, mod => %s\n%!" txt
     (if f == stub then "stub" else "real")
-    (run `Nonrec f)
-    (run `Self f)
-    (run `Mod f)
+    (run `Nonrec f) (run `Self f) (run `Mod f)
 
 module rec M : sig
   val f1 : [ `Nonrec | `Self | `Mod ] -> int
diff --git a/ppx/ppx_js/lib_internal/ppx_js_internal.ml b/ppx/ppx_js/lib_internal/ppx_js_internal.ml
index 404ddf08e6..2a5a874b0b 100644
--- a/ppx/ppx_js/lib_internal/ppx_js_internal.ml
+++ b/ppx/ppx_js/lib_internal/ppx_js_internal.ml
@@ -374,8 +374,7 @@ let prop_set ~loc ~prop_loc obj prop value =
             let loc = !default_loc in
             assert (Arg.label obj = nolabel);
             assert (Arg.label arg = nolabel);
-            arrows
-              [ (nolabel, Arg.typ obj) ]
+            arrows [ (nolabel, Arg.typ obj) ]
               (Js.type_ "gen_prop"
                  [ [%type: < set : [%t Arg.typ arg] -> unit ; .. > ] ])
         | _ -> assert false)
diff --git a/src/dune_engine/action_to_sh.ml b/src/dune_engine/action_to_sh.ml
index f3438edda..49e49ab23 100644
--- a/src/dune_engine/action_to_sh.ml
+++ b/src/dune_engine/action_to_sh.ml
@@ -22,8 +22,7 @@ let echo s =
     match List.rev lines with
     | [] -> [ Run ("echo", [ "-n" ]) ]
     | last :: rest ->
-        List.fold_left rest
-          ~init:[ Run ("echo", [ "-n"; last ]) ]
+        List.fold_left rest ~init:[ Run ("echo", [ "-n"; last ]) ]
           ~f:(fun acc s -> Run ("echo", [ s ]) :: acc)
   else List.map lines ~f:(fun s -> Run ("echo", [ s ]))
 
diff --git a/src/dune_engine/process.ml b/src/dune_engine/process.ml
index bbb39023c..b13f2ffee 100644
--- a/src/dune_engine/process.ml
+++ b/src/dune_engine/process.ml
@@ -364,8 +364,7 @@ end = struct
               [
                 (has_unexpected_stdout, "stdout");
                 (has_unexpected_stderr, "stderr");
-              ]
-              ~f:(fun (b, name) -> Option.some_if b name)
+              ] ~f:(fun (b, name) -> Option.some_if b name)
           in
           match (n, unexpected_outputs) with
           | 0, _ :: _ ->
diff --git a/src/dune_rules/dune_file.ml b/src/dune_rules/dune_file.ml
index be1288d06..1f2460d6f 100644
--- a/src/dune_rules/dune_file.ml
+++ b/src/dune_rules/dune_file.ml
@@ -1125,8 +1125,7 @@ module Executables = struct
           let files =
             List.map2 t.names public_names ~f:(fun (locn, name) (locp, pub) ->
                 Option.map pub ~f:(fun pub ->
-                    File_binding.Unexpanded.make
-                      ~src:(locn, name ^ ext)
+                    File_binding.Unexpanded.make ~src:(locn, name ^ ext)
                       ~dst:(locp, pub)))
             |> List.filter_opt
           in
@@ -1733,9 +1732,7 @@ module Rule = struct
             Static
               {
                 targets =
-                  List.map
-                    [ name ^ ".ml"; name ^ ".mli" ]
-                    ~f:(fun target ->
+                  List.map [ name ^ ".ml"; name ^ ".mli" ] ~f:(fun target ->
                       (S.make_text loc target, Targets_spec.Kind.File));
                 multiplicity = Multiple;
               };
diff --git a/src/dune_rules/findlib/meta.ml b/src/dune_rules/findlib/meta.ml
index 19ad06557..19a0ef294 100644
--- a/src/dune_rules/findlib/meta.ml
+++ b/src/dune_rules/findlib/meta.ml
diff --git a/src/dune_engine/action_to_sh.ml b/src/dune_engine/action_to_sh.ml
index f3438edda..49e49ab23 100644
--- a/src/dune_engine/action_to_sh.ml
+++ b/src/dune_engine/action_to_sh.ml
@@ -22,8 +22,7 @@ let echo s =
     match List.rev lines with
     | [] -> [ Run ("echo", [ "-n" ]) ]
     | last :: rest ->
-        List.fold_left rest
-          ~init:[ Run ("echo", [ "-n"; last ]) ]
+        List.fold_left rest ~init:[ Run ("echo", [ "-n"; last ]) ]
           ~f:(fun acc s -> Run ("echo", [ s ]) :: acc)
   else List.map lines ~f:(fun s -> Run ("echo", [ s ]))
 
diff --git a/src/dune_engine/process.ml b/src/dune_engine/process.ml
index bbb39023c..b13f2ffee 100644
--- a/src/dune_engine/process.ml
+++ b/src/dune_engine/process.ml
@@ -364,8 +364,7 @@ end = struct
               [
                 (has_unexpected_stdout, "stdout");
                 (has_unexpected_stderr, "stderr");
-              ]
-              ~f:(fun (b, name) -> Option.some_if b name)
+              ] ~f:(fun (b, name) -> Option.some_if b name)
           in
           match (n, unexpected_outputs) with
           | 0, _ :: _ ->
diff --git a/src/dune_rules/dune_file.ml b/src/dune_rules/dune_file.ml
index be1288d06..1f2460d6f 100644
--- a/src/dune_rules/dune_file.ml
+++ b/src/dune_rules/dune_file.ml
@@ -1125,8 +1125,7 @@ module Executables = struct
           let files =
             List.map2 t.names public_names ~f:(fun (locn, name) (locp, pub) ->
                 Option.map pub ~f:(fun pub ->
-                    File_binding.Unexpanded.make
-                      ~src:(locn, name ^ ext)
+                    File_binding.Unexpanded.make ~src:(locn, name ^ ext)
diff --git a/src/dune_engine/action_to_sh.ml b/src/dune_engine/action_to_sh.ml
index f3438edda..49e49ab23 100644
--- a/src/dune_engine/action_to_sh.ml
+++ b/src/dune_engine/action_to_sh.ml
@@ -22,8 +22,7 @@ let echo s =
     match List.rev lines with
     | [] -> [ Run ("echo", [ "-n" ]) ]
     | last :: rest ->
-        List.fold_left rest
-          ~init:[ Run ("echo", [ "-n"; last ]) ]
+        List.fold_left rest ~init:[ Run ("echo", [ "-n"; last ]) ]
           ~f:(fun acc s -> Run ("echo", [ s ]) :: acc)
   else List.map lines ~f:(fun s -> Run ("echo", [ s ]))
 
diff --git a/src/dune_engine/process.ml b/src/dune_engine/process.ml
index bbb39023c..b13f2ffee 100644
--- a/src/dune_engine/process.ml
+++ b/src/dune_engine/process.ml
@@ -364,8 +364,7 @@ end = struct
               [
                 (has_unexpected_stdout, "stdout");
                 (has_unexpected_stderr, "stderr");
-              ]
-              ~f:(fun (b, name) -> Option.some_if b name)
+              ] ~f:(fun (b, name) -> Option.some_if b name)
           in
           match (n, unexpected_outputs) with
           | 0, _ :: _ ->
diff --git a/src/dune_rules/dune_file.ml b/src/dune_rules/dune_file.ml
index be1288d06..1f2460d6f 100644
--- a/src/dune_rules/dune_file.ml
+++ b/src/dune_rules/dune_file.ml
@@ -1125,8 +1125,7 @@ module Executables = struct
           let files =
             List.map2 t.names public_names ~f:(fun (locn, name) (locp, pub) ->
                 Option.map pub ~f:(fun pub ->
-                    File_binding.Unexpanded.make
-                      ~src:(locn, name ^ ext)
+                    File_binding.Unexpanded.make ~src:(locn, name ^ ext)
                       ~dst:(locp, pub)))
             |> List.filter_opt
           in
@@ -1733,9 +1732,7 @@ module Rule = struct
             Static
               {
                 targets =
-                  List.map
-                    [ name ^ ".ml"; name ^ ".mli" ]
-                    ~f:(fun target ->
+                  List.map [ name ^ ".ml"; name ^ ".mli" ] ~f:(fun target ->
                       (S.make_text loc target, Targets_spec.Kind.File));
                 multiplicity = Multiple;
               };
diff --git a/src/dune_rules/findlib/meta.ml b/src/dune_rules/findlib/meta.ml
index 19ad06557..19a0ef294 100644
--- a/src/dune_engine/process.ml
+++ b/src/dune_engine/process.ml
@@ -364,8 +364,7 @@ end = struct
               [
                 (has_unexpected_stdout, "stdout");
                 (has_unexpected_stderr, "stderr");
-              ]
-              ~f:(fun (b, name) -> Option.some_if b name)
+              ] ~f:(fun (b, name) -> Option.some_if b name)
           in
           match (n, unexpected_outputs) with
           | 0, _ :: _ ->
diff --git a/src/dune_rules/dune_file.ml b/src/dune_rules/dune_file.ml
index be1288d06..1f2460d6f 100644
--- a/src/dune_rules/dune_file.ml
+++ b/src/dune_rules/dune_file.ml
@@ -1125,8 +1125,7 @@ module Executables = struct
           let files =
             List.map2 t.names public_names ~f:(fun (locn, name) (locp, pub) ->
                 Option.map pub ~f:(fun pub ->
-                    File_binding.Unexpanded.make
-                      ~src:(locn, name ^ ext)
+                    File_binding.Unexpanded.make ~src:(locn, name ^ ext)
                       ~dst:(locp, pub)))
             |> List.filter_opt
           in
@@ -1733,9 +1732,7 @@ module Rule = struct
             Static
               {
                 targets =
-                  List.map
-                    [ name ^ ".ml"; name ^ ".mli" ]
-                    ~f:(fun target ->
+                  List.map [ name ^ ".ml"; name ^ ".mli" ] ~f:(fun target ->
                       (S.make_text loc target, Targets_spec.Kind.File));
                 multiplicity = Multiple;
               };
diff --git a/src/dune_rules/findlib/meta.ml b/src/dune_rules/findlib/meta.ml
index 19ad06557..19a0ef294 100644
--- a/src/dune_rules/findlib/meta.ml
+++ b/src/dune_rules/findlib/meta.ml
@@ -143,8 +143,7 @@ let archives ?(kind = [ Mode.Byte; Mode.Native ]) name =
       (Mode.Native, archive, Mode.compiled_lib_ext);
       (Mode.Byte, plugin, Mode.compiled_lib_ext);
       (Mode.Native, plugin, Mode.plugin_ext);
-    ]
-    ~f:(fun (k, f, ext) ->
+    ] ~f:(fun (k, f, ext) ->
       if List.mem kind k ~equal:Mode.equal then
         Some (f (Mode.to_string k) (name ^ ext k))
       else None)
diff --git a/src/dune_rules/merlin.ml b/src/dune_rules/merlin.ml
index e5faf75dc..350be031f 100644
--- a/src/dune_rules/merlin.ml
+++ b/src/dune_rules/merlin.ml
@@ -170,14 +170,15 @@ module Processed = struct
                 init.config.src_dirs,
                 [ init.config.flags ],
                 init.config.extensions )
-            ~f:
-              (fun (acc_pp, acc_obj, acc_src, acc_flags, acc_ext)
-                   {
-                     modules = _;
-                     pp_config;
-                     config =
-                       { stdlib_dir = _; obj_dirs; src_dirs; flags; extensions };
-                   } ->
+            ~f:(fun
+                 (acc_pp, acc_obj, acc_src, acc_flags, acc_ext)
+                 {
+                   modules = _;
+                   pp_config;
+                   config =
+                     { stdlib_dir = _; obj_dirs; src_dirs; flags; extensions };
+                 }
+               ->
               ( pp_config :: acc_pp,
                 Path.Set.union acc_obj obj_dirs,
                 Path.Set.union acc_src src_dirs,
diff --git a/asmcomp/cmm_helpers.ml b/asmcomp/cmm_helpers.ml
index 7860bbecc..6b3e1408a 100644
--- a/asmcomp/cmm_helpers.ml
+++ b/asmcomp/cmm_helpers.ml
@@ -1831,9 +1831,8 @@ let transl_switch_clambda loc arg index cases =
   | [ _ ] -> cases.(0)
   | inters ->
       bind "switcher" arg (fun a ->
-          SwitcherBlocks.zyva loc
-            (0, n_index - 1)
-            a (Array.of_list inters) store)
+          SwitcherBlocks.zyva loc (0, n_index - 1) a (Array.of_list inters)
+            store)
 
 let strmatch_compile =
   let module S = Strmatch.Make (struct
   | inters ->
       bind "switcher" arg (fun a ->
-          SwitcherBlocks.zyva loc
-            (0, n_index - 1)
-            a (Array.of_list inters) store)
+          SwitcherBlocks.zyva loc (0, n_index - 1) a (Array.of_list inters)
+            store)
 
 let strmatch_compile =
   let module S = Strmatch.Make (struct
diff --git a/lambda/matching.ml b/lambda/matching.ml
index 35e6c2a5f..c39d3ef02 100644
--- a/lambda/matching.ml
+++ b/lambda/matching.ml
@@ -1651,9 +1651,7 @@ let divide_variant ~scopes row ctx { cases = cl; args; default = def } =
                 (make_matching
                    (get_expr_args_variant_nonconst ~scopes)
                    head def ctx)
-                ( = ) (Cstr_block tag)
-                (pat :: patl, action)
-                variants)
+                ( = ) (Cstr_block tag) (pat :: patl, action) variants)
   in
   divide cl
 
@@ -3187,8 +3185,7 @@ let for_trywith ~scopes loc param pat_act_list =
 
 let simple_for_let ~scopes loc param pat body =
   compile_matching ~scopes loc ~failer:Raise_match_failure None param
-    [ (pat, body) ]
-    Partial
+    [ (pat, body) ] Partial
 
 (* Optimize binding of immediate tuples
 
@@ -3441,8 +3438,7 @@ let do_for_multiple_match ~scopes loc paraml pat_act_list partial =
     let partial = check_partial pat_act_list partial in
     let rows = map_on_rows (fun p -> (p, [])) pat_act_list in
     toplevel_handler ~scopes loc ~failer:Raise_match_failure partial
-      [ (arg, Strict) ]
-      rows
+      [ (arg, Strict) ] rows
   in
   handler (fun partial pm1 ->
       let pm1_half =
diff --git a/lambda/translattribute.ml b/lambda/translattribute.ml
index f6e6125a5..02fa82893 100644
--- a/lambda/translattribute.ml
+++ b/lambda/translattribute.ml
@@ -170,8 +170,7 @@ let parse_poll_attribute attr =
   | None -> Default_poll
   | Some { Parsetree.attr_name = { txt; loc }; attr_payload = payload } ->
       parse_id_payload txt loc ~default:Default_poll ~empty:Default_poll
-        [ ("error", Error_poll) ]
-        payload
+        [ ("error", Error_poll) ] payload
 
 let get_inline_attribute l =
   let attr, _ = find_attribute is_inline_attribute l in
diff --git a/lambda/translclass.ml b/lambda/translclass.ml
index 3305e64c2..b31049bb7 100644
--- a/lambda/translclass.ml
+++ b/lambda/translclass.ml
@@ -210,8 +210,7 @@ let rec build_object_init ~scopes cl_table obj params inh_init obj_init cl =
               loc = of_location ~scopes pat.pat_loc;
               body =
                 Matching.for_function ~scopes pat.pat_loc None (Lvar param)
-                  [ (pat, rem) ]
-                  partial;
+                  [ (pat, rem) ] partial;
             }
         in
         match obj_init with
@@ -504,8 +503,7 @@ let rec transl_class_rebind ~scopes obj_init cl vf =
             loc = of_location ~scopes pat.pat_loc;
             body =
               Matching.for_function ~scopes pat.pat_loc None (Lvar param)
-                [ (pat, rem) ]
-                partial;
+                [ (pat, rem) ] partial;
           }
       in
       ( path,
@@ -590,15 +588,13 @@ let transl_class_rebind ~scopes cl vf =
                 ( Pmakeblock (0, Immutable, None),
                   [
                     mkappl (Lvar new_init, [ lfield cla 0 ]);
-                    lfunction
-                      [ (table, Pgenval) ]
+                    lfunction [ (table, Pgenval) ]
                       (Llet
                          ( Strict,
                            Pgenval,
                            env_init,
                            mkappl (lfield cla 1, [ Lvar table ]),
-                           lfunction
-                             [ (envs, Pgenval) ]
+                           lfunction [ (envs, Pgenval) ]
                              (mkappl
                                 ( Lvar new_init,
                                   [ mkappl (Lvar env_init, [ Lvar envs ]) ] ))
diff --git a/lambda/translcore.ml b/lambda/translcore.ml
index a51fd3f3f..52a0210a8 100644
--- a/lambda/translcore.ml
+++ b/lambda/translcore.ml
@@ -927,8 +927,7 @@ and transl_curried_function ~scopes loc return repr partial (param : Ident.t)
               cases'
           in
           ( (Curried, (param, kind) :: params, return),
-            Matching.for_function ~scopes loc None (Lvar param)
-              [ (pat, body) ]
+            Matching.for_function ~scopes loc None (Lvar param) [ (pat, body) ]
               partial )
         else (
           (match partial with
@@ -1281,9 +1280,7 @@ and transl_match ~scopes e arg pat_expr_list partial =
     | arg, _ :: _ ->
         let val_id = Typecore.name_pattern "val" (List.map fst val_cases) in
         let k = Typeopt.value_kind arg.exp_env arg.exp_type in
-        static_catch
-          [ transl_exp ~scopes arg ]
-          [ (val_id, k) ]
+        static_catch [ transl_exp ~scopes arg ] [ (val_id, k) ]
           (Matching.for_function ~scopes e.exp_loc None (Lvar val_id) val_cases
              partial)
   in
diff --git a/lambda/translobj.ml b/lambda/translobj.ml
index 96591a402..f8352492e 100644
--- a/lambda/translobj.ml
+++ b/lambda/translobj.ml
@@ -168,8 +168,7 @@ let oo_wrap env req f x =
     if !cache_required then f x
     else Misc.protect_refs [ Misc.R (cache_required, true) ] (fun () -> f x)
   else
-    Misc.protect_refs
-      [ Misc.R (wrapping, true); Misc.R (top_env, env) ]
+    Misc.protect_refs [ Misc.R (wrapping, true); Misc.R (top_env, env) ]
       (fun () ->
         cache_required := req;
         classes := [];
diff --git a/middle_end/flambda/flambda.ml b/middle_end/flambda/flambda.ml
index b666471b1..6fcd8a59e 100644
--- a/middle_end/flambda/flambda.ml
+++ b/middle_end/flambda/flambda.ml
@@ -835,8 +835,7 @@ let fold_lets_option t ~init
           | Some defining_expr ->
               let let_expr = W.create_let_reusing_body var defining_expr t in
               (acc, W.of_expr let_expr))
-        (acc, W.of_expr last_body)
-        rev_lets
+        (acc, W.of_expr last_body) rev_lets
     in
     (W.contents t, acc)
   in
diff --git a/parsing/location.ml b/parsing/location.ml
index 0944aac90..88b1cb439 100644
--- a/parsing/location.ml
+++ b/parsing/location.ml
@@ -277,8 +277,7 @@ end = struct
           | `E, `Outside -> assert false
           | `E, `Inside (s, 0) -> (`Outside, (s, a) :: acc)
           | `E, `Inside (s, n) -> (`Inside (s, n - 1), acc))
-        (`Outside, [])
-        pos
+        (`Outside, []) pos
     in
     assert (nesting = `Outside);
     List.rev acc
diff --git a/testsuite/tests/basic-modules/recursive_module_init.ml b/testsuite/tests/basic-modules/recursive_module_init.ml
index f3e08baef..00e8f9c13 100644
--- a/testsuite/tests/basic-modules/recursive_module_init.ml
+++ b/testsuite/tests/basic-modules/recursive_module_init.ml
@@ -8,9 +8,7 @@ let check ~stub txt f =
   in
   Printf.printf "%5s[%s]: nonrec => %s, self => %s, mod => %s\n%!" txt
     (if f == stub then "stub" else "real")
-    (run `Nonrec f)
-    (run `Self f)
-    (run `Mod f)
+    (run `Nonrec f) (run `Self f) (run `Mod f)
 
 module rec M : sig
   val f1 : [ `Nonrec | `Self | `Mod ] -> int
diff --git a/testsuite/tests/basic-more/morematch.ml b/testsuite/tests/basic-more/morematch.ml
index 13e5ea00a..8209b2c67 100644
--- a/testsuite/tests/basic-more/morematch.ml
+++ b/testsuite/tests/basic-more/morematch.ml
@@ -539,34 +539,26 @@ let replicaContent2shortString rc =
       "assert false"
 ;;
 
-test "jerome_variant" replicaContent2shortString
-  (`ABSENT, `Unchanged)
+test "jerome_variant" replicaContent2shortString (`ABSENT, `Unchanged)
   "        ";
 test "jerome_variant" replicaContent2shortString (`ABSENT, `Deleted) "deleted ";
 test "jerome_variant" replicaContent2shortString (`FILE, `Modified) "changed ";
 test "jerome_variant" replicaContent2shortString
   (`DIRECTORY, `PropsChanged)
   "props   ";
-test "jerome_variant" replicaContent2shortString
-  (`FILE, `Deleted)
+test "jerome_variant" replicaContent2shortString (`FILE, `Deleted)
   "assert false";
-test "jerome_variant" replicaContent2shortString
-  (`SYMLINK, `Deleted)
+test "jerome_variant" replicaContent2shortString (`SYMLINK, `Deleted)
   "assert false";
-test "jerome_variant" replicaContent2shortString
-  (`SYMLINK, `PropsChanged)
+test "jerome_variant" replicaContent2shortString (`SYMLINK, `PropsChanged)
   "assert false";
-test "jerome_variant" replicaContent2shortString
-  (`DIRECTORY, `Deleted)
+test "jerome_variant" replicaContent2shortString (`DIRECTORY, `Deleted)
   "assert false";
-test "jerome_variant" replicaContent2shortString
-  (`ABSENT, `Created)
+test "jerome_variant" replicaContent2shortString (`ABSENT, `Created)
   "assert false";
-test "jerome_variant" replicaContent2shortString
-  (`ABSENT, `Modified)
+test "jerome_variant" replicaContent2shortString (`ABSENT, `Modified)
   "assert false";
-test "jerome_variant" replicaContent2shortString
-  (`ABSENT, `PropsChanged)
+test "jerome_variant" replicaContent2shortString (`ABSENT, `PropsChanged)
   "assert false"
 
 (* bug 319 *)
diff --git a/toplevel/toploop.ml b/toplevel/toploop.ml
index d5b38fdd7..601efbc61 100644
--- a/toplevel/toploop.ml
+++ b/toplevel/toploop.ml
@@ -92,9 +92,8 @@ let use_input ppf name = use_input ppf ~wrap_in_module:false name
 let use_file ppf name = use_input ppf (File name)
 
 let use_silently ppf name =
-  Misc.protect_refs
-    [ R (use_print_results, false) ]
-    (fun () -> use_input ppf name)
+  Misc.protect_refs [ R (use_print_results, false) ] (fun () ->
+      use_input ppf name)
 
 let load_file = load_file false
 
diff --git a/typing/ctype.ml b/typing/ctype.ml
index ea1a5b88f..165cd939d 100644
--- a/typing/ctype.ml
+++ b/typing/ctype.ml
@@ -4454,9 +4454,8 @@ let rec build_subtype env (visited : transient_expr list)
             let loops = (get_id ty, t'') :: loops in
             (* May discard [visited] as level is going down *)
             let ty1', c =
-              build_subtype env
-                [ Transient_expr.repr t' ]
-                loops posi (pred_enlarge level') ty1
+              build_subtype env [ Transient_expr.repr t' ] loops posi
+                (pred_enlarge level') ty1
             in
             assert (is_Tvar t'');
             let nm =
diff --git a/typing/persistent_env.ml b/typing/persistent_env.ml
index cf70f951c..d9161d169 100644
--- a/typing/persistent_env.ml
+++ b/typing/persistent_env.ml
@@ -137,9 +137,8 @@ let without_cmis penv f x =
   let log = Lazy_backtrack.log () in
   let res =
     Misc.(
-      protect_refs
-        [ R (penv.can_load_cmis, Cannot_load_cmis log) ]
-        (fun () -> f x))
+      protect_refs [ R (penv.can_load_cmis, Cannot_load_cmis log) ] (fun () ->
+          f x))
   in
   Lazy_backtrack.backtrack log;
   res
diff --git a/typing/printtyp.ml b/typing/printtyp.ml
index 4edbe7cd6..47cca78ca 100644
--- a/typing/printtyp.ml
+++ b/typing/printtyp.ml
@@ -2420,6 +2420,5 @@ let type_expansion mode ppf ty_exp =
   type_expansion ppf (trees_of_type_expansion mode ty_exp)
 
 let tree_of_type_declaration ident td rs =
-  with_hidden_items
-    [ { hide = true; ident } ]
-    (fun () -> tree_of_type_declaration ident td rs)
+  with_hidden_items [ { hide = true; ident } ] (fun () ->
+      tree_of_type_declaration ident td rs)
diff --git a/typing/typecore.ml b/typing/typecore.ml
index afba16655..bfa58ff43 100644
--- a/typing/typecore.ml
+++ b/typing/typecore.ml
@@ -2171,9 +2171,7 @@ and type_pat_aux :
 
 let type_pat category ?no_existentials ?(mode = Normal)
     ?(lev = get_current_level ()) env sp expected_ty =
-  Misc.protect_refs
-    [ Misc.R (gadt_equations_level, Some lev) ]
-    (fun () ->
+  Misc.protect_refs [ Misc.R (gadt_equations_level, Some lev) ] (fun () ->
       type_pat category ~no_existentials ~mode ~env sp expected_ty (fun x -> x))
 
 (* this function is passed to Partial.parmatch
@@ -3131,12 +3129,10 @@ and type_expect_ ?in_function ?(recarg = Rejected) env sexp
       let body =
         Exp.let_ ~loc Nonrecursive
           ~attrs:[ Attr.mk (mknoloc "#default") (PStr []) ]
-          [ Vb.mk spat smatch ]
-          sbody
+          [ Vb.mk spat smatch ] sbody
       in
       type_function ?in_function loc sexp.pexp_attributes env
-        ty_expected_explained l
-        [ Exp.case pat body ]
+        ty_expected_explained l [ Exp.case pat body ]
   | Pexp_fun (l, None, spat, sbody) ->
       type_function ?in_function loc sexp.pexp_attributes env
         ty_expected_explained l
diff --git a/typing/typetexp.ml b/typing/typetexp.ml
index 920303d0e..c78335491 100644
--- a/typing/typetexp.ml
+++ b/typing/typetexp.ml
@@ -407,9 +407,8 @@ and transl_type_aux env policy styp =
       let mkfield l f =
         newty
           (Tvariant
-             (create_row
-                ~fields:[ (l, f) ]
-                ~more:(newvar ()) ~closed:true ~fixed:None ~name:None))
+             (create_row ~fields:[ (l, f) ] ~more:(newvar ()) ~closed:true
+                ~fixed:None ~name:None))
       in
```

</details>

janestreet profile:
<details>

```diff
diff --git a/sledge/src/fol/arithmetic.ml b/sledge/src/fol/arithmetic.ml
index 259c69c52..999f5ee5f 100644
--- a/sledge/src/fol/arithmetic.ml
+++ b/sledge/src/fol/arithmetic.ml
@@ -379,10 +379,7 @@ struct
       (match get_mono poly with
       | Some mono ->
         let mono', (coeff, mono_delta) =
-          Prod.fold
-            mono
-            (mono, (Q.one, Mono.one))
-            ~f:(fun base power (mono', delta) ->
+          Prod.fold mono (mono, (Q.one, Mono.one)) ~f:(fun base power (mono', delta) ->
               let base' = f base in
               if base' == base
               then mono', delta
diff --git a/sledge/src/fol/context.ml b/sledge/src/fol/context.ml
index 228d7a94a..a4b341193 100644
--- a/sledge/src/fol/context.ml
+++ b/sledge/src/fol/context.ml
@@ -853,10 +853,7 @@ let inter wrt r s =
   else (
     let merge_mems rs r s =
       Trm.Map.fold s.cls rs ~f:(fun ~key:rep ~data:cls rs ->
-          Cls.fold
-            cls
-            ([ rep ], rs)
-            ~f:(fun exp (reps, rs) ->
+          Cls.fold cls ([ rep ], rs) ~f:(fun exp (reps, rs) ->
               match List.find ~f:(fun rep -> implies r (Fml.eq exp rep)) reps with
               | Some rep -> reps, and_eq ~wrt exp rep rs
               | None -> exp :: reps, rs)
diff --git a/src/dune_engine/action_to_sh.ml b/src/dune_engine/action_to_sh.ml
index ba5e0ee2c..2636971d7 100644
--- a/src/dune_engine/action_to_sh.ml
+++ b/src/dune_engine/action_to_sh.ml
@@ -26,10 +26,8 @@ let echo s =
     match List.rev lines with
     | [] -> [ Run ("echo", [ "-n" ]) ]
     | last :: rest ->
-      List.fold_left
-        rest
-        ~init:[ Run ("echo", [ "-n"; last ]) ]
-        ~f:(fun acc s -> Run ("echo", [ s ]) :: acc))
+      List.fold_left rest ~init:[ Run ("echo", [ "-n"; last ]) ] ~f:(fun acc s ->
+          Run ("echo", [ s ]) :: acc))
   else List.map lines ~f:(fun s -> Run ("echo", [ s ]))
 ;;
 
diff --git a/src/dune_rules/dune_file.ml b/src/dune_rules/dune_file.ml
index 62a842141..ed708d065 100644
--- a/src/dune_rules/dune_file.ml
+++ b/src/dune_rules/dune_file.ml
@@ -1841,9 +1841,8 @@ module Rule = struct
         { targets =
             Static
               { targets =
-                  List.map
-                    [ name ^ ".ml"; name ^ ".mli" ]
-                    ~f:(fun target -> S.make_text loc target, Targets_spec.Kind.File)
+                  List.map [ name ^ ".ml"; name ^ ".mli" ] ~f:(fun target ->
+                      S.make_text loc target, Targets_spec.Kind.File)
               ; multiplicity = Multiple
               }
         ; deps = Bindings.singleton (Dep_conf.File (S.virt_text __POS__ src))
diff --git a/src/dune_rules/merlin.ml b/src/dune_rules/merlin.ml
index c758768e9..0e3672027 100644
--- a/src/dune_rules/merlin.ml
+++ b/src/dune_rules/merlin.ml
@@ -179,12 +179,13 @@ module Processed = struct
             , init.config.src_dirs
             , [ init.config.flags ]
             , init.config.extensions )
-          ~f:
-            (fun (acc_pp, acc_obj, acc_src, acc_flags, acc_ext)
-                 { modules = _
-                 ; pp_config
-                 ; config = { stdlib_dir = _; obj_dirs; src_dirs; flags; extensions }
-                 } ->
+          ~f:(fun
+               (acc_pp, acc_obj, acc_src, acc_flags, acc_ext)
+               { modules = _
+               ; pp_config
+               ; config = { stdlib_dir = _; obj_dirs; src_dirs; flags; extensions }
+               }
+             ->
             ( pp_config :: acc_pp
             , Path.Set.union acc_obj obj_dirs
             , Path.Set.union acc_src src_dirs
diff --git a/lambda/translobj.ml b/lambda/translobj.ml
index 9bfb7218f..d7a76f579 100644
--- a/lambda/translobj.ml
+++ b/lambda/translobj.ml
@@ -176,9 +176,7 @@ let oo_wrap env req f x =
     then f x
     else Misc.protect_refs [ Misc.R (cache_required, true) ] (fun () -> f x)
   else
-    Misc.protect_refs
-      [ Misc.R (wrapping, true); Misc.R (top_env, env) ]
-      (fun () ->
+    Misc.protect_refs [ Misc.R (wrapping, true); Misc.R (top_env, env) ] (fun () ->
         cache_required := req;
         classes := [];
         method_ids := Ident.Set.empty;
diff --git a/typing/printtyp.ml b/typing/printtyp.ml
index bc6c1f7f9..270e12f81 100644
--- a/typing/printtyp.ml
+++ b/typing/printtyp.ml
@@ -2695,7 +2695,6 @@ let type_expansion mode ppf ty_exp =
 ;;
 
 let tree_of_type_declaration ident td rs =
-  with_hidden_items
-    [ { hide = true; ident } ]
-    (fun () -> tree_of_type_declaration ident td rs)
+  with_hidden_items [ { hide = true; ident } ] (fun () ->
+      tree_of_type_declaration ident td rs)
 ;;
diff --git a/typing/typecore.ml b/typing/typecore.ml
index 799149c24..3b6e7502d 100644
--- a/typing/typecore.ml
+++ b/typing/typecore.ml
@@ -2271,9 +2271,8 @@ let type_pat
     sp
     expected_ty
   =
-  Misc.protect_refs
-    [ Misc.R (gadt_equations_level, Some lev) ]
-    (fun () -> type_pat category ~no_existentials ~mode ~env sp expected_ty (fun x -> x))
+  Misc.protect_refs [ Misc.R (gadt_equations_level, Some lev) ] (fun () ->
+      type_pat category ~no_existentials ~mode ~env sp expected_ty (fun x -> x))
 ;;
```

</details>

Reply:
Looks good to me.

Reply:
I prefer when "not simple" expression are highlighted (eg. nothing comes after it on the same line), even if this concept of "simple" is a bit vague.
I think this new rule is a big change and should be discussed with more people.

Reply:
I have split this PR in 2, #1968 carries out the changes for variants, while we will keep working on the lists and variants on this one.

Reply:
I have revised this PR to only make empty lists/arrays and singleton lists/arrays to be "trivial", without changing the "simple" definition, the resulting diff when running test_branch.sh is:

<details>

```diff
diff --git a/infer/src/clang/cTrans.ml b/infer/src/clang/cTrans.ml
index 57d35d743..897992c82 100644
--- a/infer/src/clang/cTrans.ml
+++ b/infer/src/clang/cTrans.ml
@@ -3243,8 +3243,7 @@ struct
             match Procdesc.Node.get_kind try_end with
             | Stmt_node ReturnStmt -> ()
             | _ ->
-                Procdesc.set_succs try_end
-                  ~normal:(Some [ try_exit_node ])
+                Procdesc.set_succs try_end ~normal:(Some [ try_exit_node ])
                   ~exn:None)
           try_ends;
         {
diff --git a/infer/src/nullsafe/unit/ThirdPartyAnnotationInfoTests.ml b/infer/src/nullsafe/unit/ThirdPartyAnnotationInfoTests.ml
index eaf2c62c5..842def6ce 100644
--- a/infer/src/nullsafe/unit/ThirdPartyAnnotationInfoTests.ml
+++ b/infer/src/nullsafe/unit/ThirdPartyAnnotationInfoTests.ml
@@ -88,8 +88,8 @@ let basic_find =
   (* Make sure we can find what we just stored *)
   assert_has_nullability_info storage
   assert_has_nullability_info storage
     { class_name = "a.A"; method_name = Method "foo"; param_types = [ "b.B" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"test.sig" ~expected_line:1;
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"test.sig"
+    ~expected_line:1;
   assert_has_nullability_info storage
     {
       class_name = "b.B";
@@ -133,8 +133,8 @@ let disregards_whitespace_lines_and_comments =
   in
   assert_has_nullability_info storage
     { class_name = "a.A"; method_name = Method "foo"; param_types = [ "b.B" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"test.sig" ~expected_line:2;
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"test.sig"
+    ~expected_line:2;
   (* Commented out signatures should be ignored *)
   assert_no_info storage
     { class_name = "a.A"; method_name = Method "bar"; param_types = [ "b.B" ] }
@@ -260,8 +260,8 @@ let can_add_several_files =
   in
   assert_has_nullability_info storage
     { class_name = "a.A"; method_name = Method "foo"; param_types = [ "b.B" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"file1.sig" ~expected_line:1;
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"file1.sig"
+    ~expected_line:1;
   (* 2. Add another file and check if we added info *)
   let file2 = [ "e.E#baz(f.F)"; "g.G#<init>(h.H, @Nullable i.I) @Nullable" ] in
   let storage =
@@ -270,13 +270,13 @@ let can_add_several_files =
   in
   assert_has_nullability_info storage
     { class_name = "e.E"; method_name = Method "baz"; param_types = [ "f.F" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"file2.sig" ~expected_line:1;
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"file2.sig"
+    ~expected_line:1;
   (* 3. Ensure we did not forget the content from the first file *)
   assert_has_nullability_info storage
     { class_name = "a.A"; method_name = Method "foo"; param_types = [ "b.B" ] }
-    ~expected_nullability:(Nonnull, [ Nonnull ])
-    ~expected_file:"file1.sig" ~expected_line:1
+    ~expected_nullability:(Nonnull, [ Nonnull ]) ~expected_file:"file1.sig"
+    ~expected_line:1
 
 let should_not_forgive_unparsable_strings =
   "should_not_forgive_unparsable_strings" >:: fun _ ->
diff --git a/infer/src/pulse/PulseCallOperations.ml b/infer/src/pulse/PulseCallOperations.ml
index ff767ed62..52901bca9 100644
--- a/infer/src/pulse/PulseCallOperations.ml
+++ b/infer/src/pulse/PulseCallOperations.ml
@@ -37,8 +37,7 @@ let unknown_call tenv path call_loc (reason : CallEvent.t) ~ret ~actuals
             List.fold fields ~init:astate ~f:(fun acc (field, field_typ, _) ->
                 let fresh_value = AbstractValue.mk_fresh () in
                 Memory.add_edge addr (FieldAccess field)
-                  (fresh_value, [ event ])
-                  call_loc acc
+                  (fresh_value, [ event ]) call_loc acc
                 |> havoc_fields (fresh_value, history) field_typ)
         | None -> astate)
     | _ -> astate
@@ -57,9 +56,8 @@ let unknown_call tenv path call_loc (reason : CallEvent.t) ~ret ~actuals
            raising issues when havoc'ing pointer parameters (which normally causes a [check_valid]
            call). *)
         let fresh_value = AbstractValue.mk_fresh () in
-        Memory.add_edge actual Dereference
-          (fresh_value, [ event ])
-          call_loc astate
+        Memory.add_edge actual Dereference (fresh_value, [ event ]) call_loc
+          astate
         |> havoc_fields actual typ
     | _ -> astate
   in
diff --git a/infer/src/pulse/PulseModels.ml b/infer/src/pulse/PulseModels.ml
index 2a702aaf4..00dea8e5c 100644
--- a/infer/src/pulse/PulseModels.ml
+++ b/infer/src/pulse/PulseModels.ml
@@ -1037,8 +1037,7 @@ module StdFunction = struct
       let empty_target = AbstractValue.mk_fresh () in
       let<+> astate =
         PulseOperations.write_deref path location ~ref:dest
-          ~obj:(empty_target, [ event ])
-          astate
+          ~obj:(empty_target, [ event ]) astate
       in
       PulseOperations.havoc_id ret_id [ event ] astate
     else
@@ -1505,14 +1504,12 @@ module JavaCollection = struct
     let init_value = AbstractValue.mk_fresh () in
     (* The two internal fields are initially set to null *)
     let<*> astate =
-      Java.write_field path fst_field
-        (init_value, [ event ])
-        location fresh_val astate
+      Java.write_field path fst_field (init_value, [ event ]) location fresh_val
+        astate
     in
     let<*> astate =
-      Java.write_field path snd_field
-        (init_value, [ event ])
-        location fresh_val astate
+      Java.write_field path snd_field (init_value, [ event ]) location fresh_val
+        astate
     in
     (* The empty field is initially set to true *)
     let<*> astate =
@@ -1581,9 +1578,8 @@ module JavaCollection = struct
     in
     let is_empty_val = AbstractValue.mk_fresh () in
     let<*> astate' =
-      Java.write_field path is_empty_field
-        (is_empty_val, [ event ])
-        location coll_val astate
+      Java.write_field path is_empty_field (is_empty_val, [ event ]) location
+        coll_val astate
     in
     (* case1: fst_field is updated *)
     let astate1 =
@@ -1626,9 +1622,8 @@ module JavaCollection = struct
     let ret_val = AbstractValue.mk_fresh () in
     let is_empty_val = AbstractValue.mk_fresh () in
     let* astate =
-      Java.write_field path is_empty_field
-        (is_empty_val, [ event ])
-        location coll_val astate
+      Java.write_field path is_empty_field (is_empty_val, [ event ]) location
+        coll_val astate
     in
     let* astate =
       PulseArithmetic.and_eq_int null_val IntLit.zero astate
@@ -1640,8 +1635,7 @@ module JavaCollection = struct
     in
     let+ astate =
       PulseOperations.write_deref path location ~ref:field_addr
-        ~obj:(null_val, [ event ])
-        astate
+        ~obj:(null_val, [ event ]) astate
     in
     PulseOperations.write_id ret_id (ret_val, [ event ]) astate
 
@@ -1715,19 +1709,16 @@ module JavaCollection = struct
       PulseOperations.eval_access path Read location coll Dereference astate
     in
     let<*> astate =
-      Java.write_field path fst_field
-        (null_val, [ event ])
-        location coll_val astate
+      Java.write_field path fst_field (null_val, [ event ]) location coll_val
+        astate
     in
     let<*> astate =
-      Java.write_field path snd_field
-        (null_val, [ event ])
-        location coll_val astate
+      Java.write_field path snd_field (null_val, [ event ]) location coll_val
+        astate
     in
     let<*> astate =
-      Java.write_field path is_empty_field
-        (is_empty_val, [ event ])
-        location coll_val astate
+      Java.write_field path is_empty_field (is_empty_val, [ event ]) location
+        coll_val astate
     in
     let<+> astate =
       PulseArithmetic.and_eq_int null_val IntLit.zero astate
@@ -1752,8 +1743,7 @@ module JavaCollection = struct
     let astate = PulseOperations.write_id ret_id (not_found_val, hist) astate in
     PulseOperations.invalidate path
       (StackAddress (Var.of_id ret_id, hist))
-      location (ConstantDereference IntLit.zero)
-      (not_found_val, [ event ])
+      location (ConstantDereference IntLit.zero) (not_found_val, [ event ])
       astate
 
   (* Auxiliary function that splits the state into three, considering the case that
diff --git a/sledge/cli/sledge_buck.ml b/sledge/cli/sledge_buck.ml
index 2d6750417..19b6ccb9f 100644
--- a/sledge/cli/sledge_buck.ml
+++ b/sledge/cli/sledge_buck.ml
@@ -95,8 +95,7 @@ let expand_arch_archive ~context archive_name =
                     (run_exit_status
                        (Lazy.force llvm_bin ^ "llvm-ar")
                        [ "p"; archive_name; name ]
-                    |+ run "head" [ "-c2" ]
-                    |+ read_all)
+                    |+ run "head" [ "-c2" ] |+ read_all)
                 in
                 if String.equal is_bc "BC" then (
                   warn "found bc file %s in %s" name archive_name ();
diff --git a/sledge/src/fol/context.ml b/sledge/src/fol/context.ml
index d10772687..1ef84e12f 100644
--- a/sledge/src/fol/context.ml
+++ b/sledge/src/fol/context.ml
@@ -758,9 +758,7 @@ let inter wrt r s =
   else
     let merge_mems rs r s =
       Trm.Map.fold s.cls rs ~f:(fun ~key:rep ~data:cls rs ->
-          Cls.fold cls
-            ([ rep ], rs)
-            ~f:(fun exp (reps, rs) ->
+          Cls.fold cls ([ rep ], rs) ~f:(fun exp (reps, rs) ->
               match
                 List.find ~f:(fun rep -> implies r (Fml.eq exp rep)) reps
               with
diff --git a/sledge/src/test/sh_test.ml b/sledge/src/test/sh_test.ml
index 72ca2699d..0cf455abb 100644
--- a/sledge/src/test/sh_test.ml
+++ b/sledge/src/test/sh_test.ml
@@ -86,8 +86,7 @@ let%test_module _ =
       let q =
         or_
           (pure (x = !0))
-          (exists
-             ~$[ x_ ]
+          (exists ~$[ x_ ]
              (or_
                 (and_ (x = !1) (pure (y = !1)))
                 (exists ~$[ x_ ] (pure (x = !2)))))
@@ -105,12 +104,10 @@ let%test_module _ =
 
     let%expect_test _ =
       let q =
-        exists
-          ~$[ x_ ]
+        exists ~$[ x_ ]
           (or_
              (pure (x = !0))
-             (exists
-                ~$[ x_ ]
+             (exists ~$[ x_ ]
                 (or_
                    (and_ (x = !1) (pure (y = !1)))
                    (exists ~$[ x_ ] (pure (x = !2))))))
@@ -130,12 +127,10 @@ let%test_module _ =
 
     let%expect_test _ =
       let q =
-        exists
-          ~$[ x_ ]
+        exists ~$[ x_ ]
           (or_
              (pure (x = !0))
-             (exists
-                ~$[ x_ ]
+             (exists ~$[ x_ ]
                 (or_
                    (and_ (x = !1) (pure (y = !1)))
                    (exists ~$[ x_ ] (pure (x = !2))))))
@@ -170,8 +165,7 @@ let%test_module _ =
              (pure (eq_concat (!16, e) [| (!8, a); (!8, d) |]))
              (or_
                 (pure (Formula.dq x !0))
-                (exists
-                   (Var.Set.of_list [ b_ ])
+                (exists (Var.Set.of_list [ b_ ])
                    (pure (eq_concat (!8, a) [| (!4, c); (!4, b) |])))))
       in
       pp_raw q;
diff --git a/sledge/src/test/solver_test.ml b/sledge/src/test/solver_test.ml
index c77dba2d8..120b44db7 100644
--- a/sledge/src/test/solver_test.ml
+++ b/sledge/src/test/solver_test.ml
@@ -105,8 +105,7 @@ let%test_module _ =
       let minued = Sh.star common seg1 in
       let subtrahend =
         Sh.and_ (Formula.eq m n)
-          (Sh.exists
-             (Var.Set.of_list [ m_ ])
+          (Sh.exists (Var.Set.of_list [ m_ ])
              (Sh.extend_us (Var.Set.of_list [ m_ ]) common))
       in
       infer_frame minued [ n_; m_ ] subtrahend;
diff --git a/src/dune_rules/merlin.ml b/src/dune_rules/merlin.ml
index e5faf75dc..350be031f 100644
--- a/src/dune_rules/merlin.ml
+++ b/src/dune_rules/merlin.ml
@@ -170,14 +170,15 @@ module Processed = struct
                 init.config.src_dirs,
                 [ init.config.flags ],
                 init.config.extensions )
-            ~f:
-              (fun (acc_pp, acc_obj, acc_src, acc_flags, acc_ext)
-                   {
-                     modules = _;
-                     pp_config;
-                     config =
-                       { stdlib_dir = _; obj_dirs; src_dirs; flags; extensions };
-                   } ->
+            ~f:(fun
+                 (acc_pp, acc_obj, acc_src, acc_flags, acc_ext)
+                 {
+                   modules = _;
+                   pp_config;
+                   config =
+                     { stdlib_dir = _; obj_dirs; src_dirs; flags; extensions };
+                 }
+               ->
               ( pp_config :: acc_pp,
                 Path.Set.union acc_obj obj_dirs,
                 Path.Set.union acc_src src_dirs,
diff --git a/src/memo/memo.ml b/src/memo/memo.ml
index 2debbc6d8..bcc074ffe 100644
--- a/src/memo/memo.ml
+++ b/src/memo/memo.ml
@@ -81,9 +81,7 @@ module Spec = struct
     let name =
       match name with
       | None when !Debug.track_locations_of_lazy_values ->
-          Option.map
-            (Caller_id.get ~skip:[ __FILE__ ])
-            ~f:(fun loc ->
+          Option.map (Caller_id.get ~skip:[ __FILE__ ]) ~f:(fun loc ->
               sprintf "lazy value created at %s" (Loc.to_file_colon_line loc))
       | _ -> name
     in
diff --git a/ocamldoc/odoc_ast.ml b/ocamldoc/odoc_ast.ml
index 75b6c45b5..feec131a4 100644
--- a/ocamldoc/odoc_ast.ml
+++ b/ocamldoc/odoc_ast.ml
@@ -564,8 +564,7 @@ functor
                     ic_text = text_opt;
                   }
                 in
-                iter
-                  (acc_inher @ [ inher ])
+                iter (acc_inher @ [ inher ])
                   (acc_fields @ ele_comments)
                   p_clexp.Parsetree.pcl_loc.Location.loc_end.Lexing.pos_cnum q
             | Parsetree.Pcf_val ({ txt = label }, mutable_flag, k) ->
diff --git a/ocamldoc/odoc_to_text.ml b/ocamldoc/odoc_to_text.ml
index c2a15d542..baaafcdd4 100644
--- a/ocamldoc/odoc_to_text.ml
+++ b/ocamldoc/odoc_to_text.ml
@@ -98,8 +98,7 @@ class virtual info =
       match l with
       | [] -> []
       | [ see ] ->
-          Bold [ Raw Odoc_messages.see_also ]
-          :: Raw " " :: self#text_of_see see
+          Bold [ Raw Odoc_messages.see_also ] :: Raw " " :: self#text_of_see see
           @ [ Newline ]
       | _ ->
           Bold [ Raw Odoc_messages.see_also ]
```

</details>

this change is far less aggressive than the previous iteration, as you can see in the diff newly "simple" expressions don't shadow the rest of the code, but it still allow more flexibility to have the code fitting on the line (like this issue's target example), what do you think? @Julow @ceastlund